### PR TITLE
8308549: Classfile API should fail to generate over-sized Code attribute

### PR DESCRIPTION
**Pull request by asotona**
Classfile API allowed to generate Code attribute exceeding the 65k limit. No exception has been thrown during class generation and the class failed verification later during class loading.
This patch adds Code size limit check throwing IllegalArgumentException.
The patch also adds similar check for constant pool size limit to avoid generation class file with corrupted  constant pool.
Two new tests are added to check response on oversized Code attribute and constant pool.
`VerifierImpl` is extended to check Code attribute size as a part of class verification process.

Please review.

Thanks,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308549](https://bugs.openjdk.org/browse/JDK-8308549): Classfile API should fail to generate over-sized Code attribute

### Reviewers
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to [32c00386](https://git.openjdk.org/jdk/pull/14100/files/32c00386e321af997037670e0db0ca6923285927)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14100/head:pull/14100` \
`$ git checkout pull/14100`

Update a local copy of the PR: \
`$ git checkout pull/14100` \
`$ git pull https://git.openjdk.org/jdk.git pull/14100/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14100`

View PR using the GUI difftool: \
`$ git pr show -t 14100`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14100.diff">https://git.openjdk.org/jdk/pull/14100.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14100#issuecomment-1559638426)

### CODE REVIEW

**Comment by liach** [comment]
On a side note, does Classfile API reject methods with too many slots (locals) (MethodTypeDesc can represent parameter lists with over 255 slots) or stack (operand)?

**Review by mlchung — APPROVED** [review]
This looks ok to me.

**Comment by asotona** [comment]
> On a side note, does Classfile API reject methods with too many slots (locals) (MethodTypeDesc can represent parameter lists with over 255 slots) or stack (operand)?

Classfile API does not perform any extra verifications of MethodTypeDesc and 255 slots is limit for method parameters. Code attribute maxLocals and maxStack have limit of 65535, however it is not checked yet.

File: src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java [hunk 1]
```diff
@@ -311,6 +311,13 @@ public void writeBody(BufWriter b) {
                 buf.setLabelContext(DirectCodeBuilder.this);
 
                 int codeLength = curPc();
+                if (codeLength >= 65536) {
```
[the remark this answers is not in the document]
Comment (asotona) [reply src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java:314]: Right, DirectCodeBuilder can be triggered empty, thanks.

[the remark this answers is not in the document]
Comment (jaikiran) [reply src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java]: Thank you Adam for the update, looks good to me.

**Pull request closed without merging**